src/toil/cwl/cwltoil.py

```diff
@@ -3574,8 +3557,73 @@ def main(args: Optional[List[str]] = None, stdout: TextIO = sys.stdout) -> int:
         type=str,
     )
 
-    # Parse all the options once.
-    options = parser.parse_args(args)
+def get_options(args: List[str]) -> argparse.Namespace:
+    """
+    Get around configargparse's incompatibility with nargs=REMAINDER
+    It does so by running parse_args twice, one on just the config file and one on the rest of args, then merges
+    Takes the list of args and returns the resulting options namespace
+    :param args: List of args from command line
+    :return: options namespace
+    """
+    parser = ArgParser()
+    addOptions(parser, jobstore_as_flag=True, cwl=True)
+    add_cwl_options(parser)
+
+    config_args = []
+    cmd_line_args = args
+    # find if the config argument is specified on the command line
+    for i in range(len(args)):
+        if args[i].startswith("--config"):
+            # if found, remove config file definition from args
+            split_by_equal = args[i].split("=")
+            if len(split_by_equal) > 1:
+                # --config=file.yaml
+                config_args = ["--config", (split_by_equal[1])]
+                cmd_line_args = args[:i] + args[i+1:]
+            elif i < len(args):
+                # --config file.yaml
+                config_args = ["--config", (args[i+1])]
+                cmd_line_args = args[:i] + args[i+2:]
+            else:
+                # not properly defined
+                # don't define the config file so that argparse can raise the proper error
+                config_args = ["--config"]
+            break
+
+    # Parse only the config file to not conflict with the cmd line options
+    config_options = parser.parse_args(config_args)
+
+    # These arguments are defined here so the previous parse_args can run
+    # This will mean that these cannot be defined in the config file and must be defined on the command line
+    parser.add_argument("cwltool", type=str)
+    parser.add_argument("cwljob", nargs=argparse.REMAINDER)
+
+    cmd_line_args_actions = {y for x in cmd_line_args for y in x.split("=")}
+    # remove parser actions that are not specified on the command line
+    i = 0
+    while i < len(parser._actions):
+        action = parser._actions[i]
+        if set(action.option_strings).isdisjoint(cmd_line_args_actions) and len(action.option_strings) > 0:
+            del parser._actions[i]
+        else:
+            i += 1
+    # Parse only cmd line args
+    cmd_line_options = parser.parse_args(cmd_line_args)
+    # Merge namespaces, with command line taking precedence over config
```

Comment:
```suggestion
    # Merge namespaces, with command line taking precedence over config.
    # Because we removed all the options not actually specified on the command line,
    # nothing will get clobbered with a default.
```